ui/components/app/qr-hardware-popover/qr-hardware-sign-request/reader.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { ETHSignature } from '@keystonehq/bc-ur-registry-eth';
+import * as uuid from 'uuid';
+import PropTypes from 'prop-types';
+import BaseReader from '../base-reader';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+
+const Reader = ({
+  submitQRHardwareSignature,
+  cancelQRHardwareSignRequest,
+  requestId,
+  setErrorTitle,
+}) => {
+  const t = useI18nContext();
+  const cancel = () => {
+    cancelQRHardwareSignRequest();
+  };
+
+  const handleSuccess = (ur) => {
+    return new Promise((resolve, reject) => {
```

Comment:
better to make this an async funcion?

Reply:
yes, will update

Reply:
done

Reply:
this was not turned into an async function